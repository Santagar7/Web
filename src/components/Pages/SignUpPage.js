//import '../../css/LoginPage.css';
import {Link, useNavigate} from "react-router-dom";
import React from 'react';
import axios from "axios";

const url = 'http://localhost:7000/';

function SignUpPage() {
    let navigator = useNavigate();
    function SignUp(event){
        event.preventDefault();
        let name = document.getElementById('name').value;
        let firstname = document.getElementById('firstname').value;
        let lastname = document.getElementById('lastname').value;
        let email = document.getElementById('email').value;
        let password = document.getElementById('password').value;
        let repeat_password = document.getElementById('repeat_password').value;

        let elem = document.getElementById('register');

        if((email.match(/@/g) || []).length!==1){
            elem.textContent = 'Invalid Email';
            elem.style.color = 'crimson';
            return;
        }
        if(password!==repeat_password){
            elem.textContent = 'Passwords should match';
            elem.style.color = 'crimson';
            return;
        }
        if(password.length<8){
            elem.textContent = 'Password is too short';
            elem.style.color = 'crimson';
            return;
        }
        if(password.replace(/[a-z]/gi, '').length<2){
            elem.textContent = 'Password not safe enough';
            elem.style.color = 'crimson';
            return;
        }

        axios.post(url+'register',{
            'name':name,
            'firstname':firstname,
            'lastname':lastname,
            'password':password,
            'email':email,
            'repeat_password':repeat_password
        }).then((resp)=>{
            if(resp.data.message==='Success'){
                navigator("/");
            }
            else{
                elem.textContent = resp.data.message;
                elem.style.color = 'crimson';
            }
        })
    }

    return (
        <div className="center">
                <h1 id="register">Registration</h1>
                <form className="main">
                    <div className="data">
                        <input type="text" id="name"  data-testid='name' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>Name</label>
                    </div>
                    <div className="data">
                        <input type="password" id="password"  data-testid='password' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>Password</label>
                    </div>
                    <div className="data">
                        <input type="password" id="repeat_password"  data-testid='repeat_password' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>Repeat Password</label>
                    </div>
                    <div className="data">
                        <input type="text" id="firstname"  data-testid='firstname' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>FirstName</label>
                    </div>
                    <div className="data">
                        <input type="text" id="lastname"  data-testid='lastname' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>LastName</label>
                    </div>
                    <div className="data">
                        <input type="text" id="email"  data-testid='email' placeholder=" " autoComplete="on"/>
                        <span/>
                        <label>Email</label>
                    </div>

                    <input type="submit" value="Submit"  data-testid='submit' onClick={SignUp}/>
                    <div className="signup">
                        <h4>Already have an account ?  <Link to='/'>Log in</Link></h4>
                    </div>
                </form>
            </div>
    );
}

export default SignUpPage;