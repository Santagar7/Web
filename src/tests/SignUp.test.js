import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SignUpPage from '../components/Pages/SignUpPage';


test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><SignUpPage/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
test('test register data', () => {
    render(
        <BrowserRouter>
            <SignUpPage />
        </BrowserRouter>,
    );
    const inputName = screen.getByTestId('name');
    fireEvent.change(inputName, { target: { value: 'steve' } });
    expect(inputName.value).toBe('steve');

    const inputPass = screen.getByTestId('password');
    fireEvent.change(inputPass, { target: { value: 'steve' } });
    expect(inputName.value).toBe('steve');

    const inputRepeatPass = screen.getByTestId('repeat_password');
    fireEvent.change(inputRepeatPass, { target: { value: 'steve12' } });
    expect(inputRepeatPass.value).toBe('steve12');

    const inputEmail = screen.getByTestId('email');
    fireEvent.change(inputEmail, { target: { value: 'email@gmail.com' } });
    expect(inputEmail.value).toBe('email@gmail.com');
});

test('Invalid Email', () => {
    render(
        <BrowserRouter>
            <SignUpPage />
        </BrowserRouter>,
    );
    const submit = screen.getByTestId('submit');
    fireEvent.click(submit);
    const linkElement = screen.getByText(/Invalid Email/i);
    expect(linkElement).toBeInTheDocument();
});
test('Passwords should match', () => {
    render(
        <BrowserRouter>
            <SignUpPage />
        </BrowserRouter>,
    );
    
    const inputEmail = screen.getByTestId('email');
    fireEvent.change(inputEmail, { target: { value: 'user@gmail.com' } });
    expect(inputEmail.value).toBe('user@gmail.com');

    const inputPass = screen.getByTestId('password');
    fireEvent.change(inputPass, { target: { value: 'steve' } });
    expect(inputPass.value).toBe('steve');

    const inputRepeatPass = screen.getByTestId('repeat_password');
    fireEvent.change(inputRepeatPass, { target: { value: 'steve12' } });
    expect(inputRepeatPass.value).toBe('steve12');

    const submit = screen.getByTestId('submit');
    fireEvent.click(submit);

    const linkElement = screen.getByText(/Passwords should match/i);
    expect(linkElement).toBeInTheDocument();
});
test('Password is too short', () => {
    render(
        <BrowserRouter>
            <SignUpPage />
        </BrowserRouter>,
    );
    
    const inputEmail = screen.getByTestId('email');
    fireEvent.change(inputEmail, { target: { value: 'user@gmail.com' } });
    expect(inputEmail.value).toBe('user@gmail.com');

    const inputPass = screen.getByTestId('password');
    fireEvent.change(inputPass, { target: { value: '123' } });
    expect(inputPass.value).toBe('123');

    const inputRepeatPass = screen.getByTestId('repeat_password');
    fireEvent.change(inputRepeatPass, { target: { value: '123' } });
    expect(inputRepeatPass.value).toBe('123');

    const submit = screen.getByTestId('submit');
    fireEvent.click(submit);

    const linkElement = screen.getByText(/Password is too short/i);
    expect(linkElement).toBeInTheDocument();
});

test('Password not safe enough', () => {
    render(
        <BrowserRouter>
            <SignUpPage />
        </BrowserRouter>,
    );
    
    const inputEmail = screen.getByTestId('email');
    fireEvent.change(inputEmail, { target: { value: 'user@gmail.com' } });
    expect(inputEmail.value).toBe('user@gmail.com');

    const inputPass = screen.getByTestId('password');
    fireEvent.change(inputPass, { target: { value: 'stevesteve' } });
    expect(inputPass.value).toBe('stevesteve');

    const inputRepeatPass = screen.getByTestId('repeat_password');
    fireEvent.change(inputRepeatPass, { target: { value: 'stevesteve' } });
    expect(inputRepeatPass.value).toBe('stevesteve');

    const submit = screen.getByTestId('submit');
    fireEvent.click(submit);

    const linkElement = screen.getByText(/Password not safe enough/i);
    expect(linkElement).toBeInTheDocument();
});