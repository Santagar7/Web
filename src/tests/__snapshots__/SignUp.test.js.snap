// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="center"
>
  <h1
    id="register"
  >
    Registration
  </h1>
  <form
    className="main"
  >
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="name"
        id="name"
        placeholder=" "
        type="text"
      />
      <span />
      <label>
        Name
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="password"
        id="password"
        placeholder=" "
        type="password"
      />
      <span />
      <label>
        Password
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="repeat_password"
        id="repeat_password"
        placeholder=" "
        type="password"
      />
      <span />
      <label>
        Repeat Password
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="firstname"
        id="firstname"
        placeholder=" "
        type="text"
      />
      <span />
      <label>
        FirstName
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="lastname"
        id="lastname"
        placeholder=" "
        type="text"
      />
      <span />
      <label>
        LastName
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="email"
        id="email"
        placeholder=" "
        type="text"
      />
      <span />
      <label>
        Email
      </label>
    </div>
    <input
      data-testid="submit"
      onClick={[Function]}
      type="submit"
      value="Submit"
    />
    <div
      className="signup"
    >
      <h4>
        Already have an account ?  
        <a
          href="/"
          onClick={[Function]}
        >
          Log in
        </a>
      </h4>
    </div>
  </form>
</div>
`;
